### ISSUE TITLE
Controller fails on empty manifest files from a Helm Chart

### ISSUE THREAD

**Issue by mphammer** [operator-framework/operator-sdk#2622]
## Bug Report

**What did you do?**
For each of the following cases I did these steps:
1. Created an Operator from a local helm chart:
`$ operator-sdk new my-operator --type=helm --helm-chart <local_helm_chart_path>`
2. Deployed the CRD
`$ kubectl create -f deploy/crds/<resource>_crd.yaml`
3. Started local operator
`$ operator-sdk run --local`
4. Create the CR
`kubectl create -f deploy/crds/<resource>_cr.yaml`

Case 1: A file containing a resource is empty due to a conditional
* myvalue is not set
```
{{ if .Values.myvalue }}
# entire resource goes here
# ...
{{ end }}
```

Case 2: There is an empty manifest from `---` markers
```
---

---
# valid resources go here
# ...
```

**What did you expect to see?**
* The controller ignores the empty resources

**What did you see instead? Under which circumstances?**
Some of the resources were deployed but I saw the following error messages repeatedly printed...
Case 1:
```
{"level":"error",
"ts":1583340099.203393,
"logger":"helm.controller",
"msg":"Release failed",
"namespace":"default",
"name":"example-product",
"apiVersion":"domain.com/v1alpha1",
"kind":"Product",
"release":"example-product",
"error":"failed to install release: no objects visited",
"stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/helm/controller.HelmOperatorReconciler.Reconcile\n\tsrc/github.com/operator-framework/operator-sdk/pkg/helm/controller/reconcile.go:194\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:88"}
```

Case 2:
```
{"level":"error",
"ts":1583340147.234184,
"logger":"helm.controller",
"msg":"Failed to run release hook",
"namespace":"default",
"name":"example-product",
"apiVersion":"domain.com/v1alpha1",
"kind":"Product",
"release":"example-product",
"error":"no matches for kind \"\" in version \"\"",
"stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/helm/controller.HelmOperatorReconciler.Reconcile\n\tsrc/github.com/operator-framework/operator-sdk/pkg/helm/controller/reconcile.go:308\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:88"}
```

**Environment**
* operator-sdk version:

```
$ operator-sdk version
operator-sdk version: "v0.15.2", commit: "ffaf278993c8fcb00c6f527c9f20091eb8dd3352", go version: "go1.13.8 darwin/amd64"
```

* go version:

```
$ go version
go version go1.13.8 darwin/amd64
````

* Kubernetes version information:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T23:43:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-gke.9", GitCommit:"a9973cbb2722793e2ea08d20880633ca61d3e669", GitTreeState:"clean", BuildDate:"2020-02-07T22:35:02Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```
* Are you writing your operator in ansible, helm, or go?
```
$ helm version
version.BuildInfo{Version:"v3.0.1", GitCommit:"7c22ef9ce89e0ebeb7125ba2ebf7d421f3e82ffa", GitTreeState:"clean", GoVersion:"go1.13.4"}
```

**Possible Solution**
* If a manifest is empty then do not attempt to reconcile it in the cluster

**Additional context**

**Comment by camilamacedo86** [comment]
Hi @mphammer, 

Could you please provide and `-helm-chart <local_helm_chart_path>` which we can use to reproduce your scenario in order to make it more clear? Have you any example from the [stable](https://github.com/helm/charts/tree/master/stable/memcached) repo which would allow us to check/reproduce it? Also, what do you mean with `https://github.com/helm/charts/tree/master/stable/memcached`? Could you please elaborate more?

**Comment by joelanford** [comment]
@mphammer 

Case 1 is actually pretty common in Helm charts. In fact, we coincidentally test this case in the SDK's helm-operator e2e test (check out the ingress template generated from [this command](https://github.com/operator-framework/operator-sdk/blob/5104f530ea7bbfb50f4d2d5888cbdc2f26786906/hack/tests/e2e-helm.sh#L129-L133) in the test).

So I'm fairly certain the Helm operator supports disabling entire resources. Your error seems to indicate that the release contained no resources at all. And in that case, it would make sense to fail an installation.

For case 2, this looks like a legitimate bug. My guess is that it is originating from [here](https://github.com/operator-framework/operator-sdk/blob/ad1d4d1953abf22c336f67198fc23a489c0f4fdb/pkg/helm/controller/controller.go#L127).

We parse each YAML document in the release manifest and assume that we'll find the object meta. In the case of your chart it looks like there are some empty documents, which parse "successfully", but result in an empty object.

Any chance you would be interested in submitting a PR to add some extra sanity checks in that function to make sure that we have a non-empty GVK?

**Comment by mphammer** [comment]
> Hi @mphammer,
> 
> Could you please provide and `-helm-chart <local_helm_chart_path>` which we can use to reproduce your scenario in order to make it more clear? Have you any example from the [stable](https://github.com/helm/charts/tree/master/stable) repo which would allow us to check/reproduce it? Also, what do you mean with `If a manifest is empty then do not attempt to reconcile it in the cluster`? Could you please elaborate more?

Hi @camilamacedo86, 

Sorry for the late response. These errors were not from the stable repo but from a custom helm chart. Here are examples of the Charts:
**Case 1:**
Chart.yaml:
```
apiVersion: v1
name: my-helm-chart
description: A helm chart for me
type: application
version: 1.0.0
appVersion: 2.3.0
```
Values.yaml:
```
resourceProvided: true
```
templates/resource.yaml:
```
{{ if .Values.resourceProvided }}
apiVersion: v1
kind: Service
metadata:
  labels:
  name: {{ .Release.Name }}-service
  namespace: {{ .Release.Namespace }}
spec:
  ports:
    - name: port-8443
      port: 8443
      protocol: TCP
      targetPort: 8443
  selector:
    name: {{ .Release.Name }}
  type: ClusterIP
{{ end }}
```

**Case 2:**
Chart.yaml:
```
apiVersion: v1
name: my-helm-chart
description: A helm chart for me
type: application
version: 1.0.0
appVersion: 2.3.0
```
Values.yaml:
```
exposeService: true
```
templates/resource.yaml:
```
apiVersion: v1
kind: Service
metadata:
  labels:
    name: {{ .Release.Name }}-label
  name: {{ .Release.Name }}-service
  namespace: {{ .Release.Namespace }}
spec:
  ports:
    - name: port-8443
      port: 8443
      protocol: TCP
      targetPort: 8443
  selector:
    name: {{ .Release.Name }}-label
  type: ClusterIP
---
{{ if .Values.exposeService }}
apiVersion: v1
kind: Service
metadata:
  labels:
    name: {{ .Release.Name }}-exposed-label
  name: {{ .Release.Name }}-service-exposed
  namespace: {{ .Release.Namespace }}
spec:
  ports:
    - name: port-8443
      port: 8443
      protocol: TCP
      targetPort: 8443
  selector:
    name: {{ .Release.Name }}-exposed-label
  type: LoadBalancer
{{ end }}
---
```

**Comment by mphammer** [comment]
Hi @joelanford, thanks for the information and the "kind/bug" label. I don't currently have the cycles to work on this issue. Thanks @bharathi-tenneti for taking this up! If you need more information from me I will do my best to assist you.

**Comment by bharathi-tenneti** [comment]
@mphammer 
When trying to reproduce CASE 1, with below environment, I see that operator shows error in the creation step itself, as shown below.
```
$ operator-sdk new my-operator --type=helm --helm-chart=my-helm-chart/
INFO[0000] Creating new Helm operator 'my-operator'.    
INFO[0000] Created helm-charts/my-helm-chart            
INFO[0000] Generating RBAC rules                        
WARN[0000] Using default RBAC rules: failed to generate RBAC rules: failed to get default manifest: failed to render chart templates: template: my-helm-chart/templates/tests/test-connection.yaml:14:65: executing "my-helm-chart/templates/tests/test-connection.yaml" at <.Values.service.port>: nil pointer evaluating interface {}.port 
INFO[0000] Created build/Dockerfile                     
INFO[0000] Created watches.yaml                         
INFO[0000] Created deploy/service_account.yaml          
INFO[0000] Created deploy/role.yaml                     
INFO[0000] Created deploy/role_binding.yaml             
INFO[0000] Created deploy/operator.yaml                 
INFO[0000] Created deploy/crds/charts.helm.k8s.io_v1alpha1_myhelmchart_cr.yaml 
INFO[0000] Generated CustomResourceDefinition manifests. 
INFO[0000] Project creation complete.                   
```
 Was curious, whether you have seen similar error while creation.
Later, encountered bellow error , after deployment of `cr`
```
{"level":"info","ts":1586186301.702219,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"myhelmchart-controller"}
{"level":"info","ts":1586186301.7022479,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"myhelmchart-controller","worker count":1}
{"level":"error","ts":1586186373.333587,"logger":"helm.controller","msg":"Release failed","namespace":"default","name":"example-myhelmchart","apiVersion":"charts.helm.k8s.io/v1alpha1","kind":"MyHelmChart","release":"example-myhelmchart","error":"failed to install release: template: my-helm-chart/templates/tests/test-connection.yaml:14:65: executing \"my-helm-chart/templates/tests/test-connection.yaml\" at <.Values.service.port>: nil pointer evaluating interface {}.port","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/helm/controller.HelmOperatorReconciler.Reconcile\n\tsrc/github.com/operator-framework/operator-sdk/pkg/helm/controller/reconcile.go:196\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\tpkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\tpkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:88"}
```

P.S 
```
operator-sdk version
operator-sdk version: "v0.16.0", commit: "55f1446c5f472e7d8e308dcdf36d0d7fc44fc4fd", go version: "go1.14 darwin/amd64"
```
```
helm version
version.BuildInfo{Version:"v3.1.0", GitCommit:"b29d20baf09943e134c2fa5e1e1cab3bf93315fa", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

**Comment by joelanford** [comment]
@mphammer @bharathi-tenneti Just want to re-iterate [my previous comment](https://github.com/operator-framework/operator-sdk/issues/2622#issuecomment-604225508) that I think something slightly different is happening with case 1 than what is described in the issue.

The error message you got (`failed to install release: no objects visited`) is actually coming from the Helm libraries [here](https://github.com/helm/helm/blob/c12a9aee02ec07b78dce07274e4816d9863d765e/pkg/kube/client.go#L313-L315). That error means that there were 0 resources in the release manifest, so there must be something wrong with your templates or CR values that is causing nothing to be rendered.

I think we can fix case 2 by checking to make sure the GVK is not empty.

**Comment by jsm84** [comment]
@joelanford  we have a partner seeking operator certification that is also seeing case #2. In their particular case, the chart installs and also renders fine using helm3. Putting the same chart into an operator yields case #2, which we believe is when the deployment templates get rendered.

The following conditional code appears to be tripping up the helm-operator:

```
{{/* Generates deployment specs for CR controller services. These are stateless. */}}
{{- range include "k10.crServices" . | splitList " " }}
  {{ $service := print "crc-" . }}
  {{ dict "main" $ "k10_service" $service "persistence.enabled" false "replicas" 1 | include "k10-default" }}
{{- end }}
```

The chart name is `k10` and can be installed from the repo at https://charts.kasten.io.

**Comment by bharathi-tenneti** [comment]
@jsm84  Can you please recheck the charts link from above, will try to reproduce.

**Comment by jsm84** [comment]
@bharathi-tenneti Sorry about that. That link is a helm repo, and not browser accessible.

The following helm commands will add the repo, and then fetch the chart in tar.gz format:

```
$ helm repo add kasten https://charts.kasten.io
$ helm fetch kasten/k10
```

**Comment by bharathi-tenneti** [comment]
@jsm84 What is your helm version and sdk version?

**Comment by depohmel** [comment]
> @jsm84 What is your helm version and sdk version?

helm chart is helm2 
sdk is 0.15.2 (0.16.0 requires h3 chart) 

this chart works find with helm3 and helm2

**Comment by joelanford** [comment]
> 0.16.0 requires h3 chart

Really? If so, that was an unexpected breakage. Do you have details?

**Comment by depohmel** [comment]
> > 0.16.0 requires h3 chart
> 
> @depohmel Really? If so, that was an unexpected breakage. Do you have details?

Unfortunately yes, it looks for Charts.lock with is available only in helm 3 since requirements were moved to Chart.yaml. I can create the issue tomorrow with details.

**Comment by depohmel** [comment]
> > 0.16.0 requires h3 chart
> 
> @depohmel Really? If so, that was an unexpected breakage. Do you have details?

FIled https://github.com/operator-framework/operator-sdk/issues/2806

**Comment by bharathi-tenneti** [comment]
@depohmel Can you please try this repo for SDK, and let us know if it fixes Case 2 for you?
https://github.com/bharathi-tenneti/operator-sdk/tree/issue%232622

**Comment by depohmel** [comment]
> @depohmel @mphammer Can you please try this repo for SDK, and let us know if it fixes Case 2 for you?
> https://github.com/bharathi-tenneti/operator-sdk/tree/issue%232622

@bharathi-tenneti  i hope ecerytihg right to build operator-sdk and package it 
but i still getting ```2020-04-09T00:10:33.996Z	DEBUG	predicates	Reconciling due to dependent resource update	{"name": "prometheus-server", "namespace": "kasten", "apiVersion": "v1", "kind": "PersistentVolumeClaim"}
2020-04-09T00:10:38.214Z	ERROR	helm.controller	Failed to run release hook	{"namespace": "kasten", "name": "k10", "apiVersion": "apik10.kasten.io/v1alpha1", "kind": "K10", "release": "k10", "error": "no matches for kind \"\" in version \"\""}
github.com/go-logr/zapr.(*zapLogger).Error
	pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
github.com/operator-framework/operator-sdk/pkg/helm/controller.HelmOperatorReconciler.Reconcile
	src/github.com/operator-framework/operator-sdk/pkg/helm/controller/reconcile.go:210
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:88
2020-04-09T00:10:38.215Z	ERROR	controller-runtime.controller	Reconciler error	{"controller": "k10-controller", "request": "kasten/k10", "error": "no matches for kind \"\" in version \"\""}
github.com/go-logr/zapr.(*zapLogger).Error
	pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:258
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	pkg/mod/k8s.io/apimachinery@v0.0.0-20191004115801-a2eda9f80ab8/pkg/util/wait/wait.go:88
2020-04-09T00:10:39.215Z	DEBUG	helm.controller	Reconciling	{"namespace": "kasten", "name": "k10", "apiVersion": "apik10.kasten.io/v1alpha1", "kind": "K10"}
E0409 00:10:40.351418       1 memcache.go:199] couldn't get resource list for actions.kio.kasten.io/v1alpha1: the server is currently unable to handle the request
```

**Comment by depohmel** [comment]
> @depohmel @mphammer Can you please try this repo for SDK, and let us know if it fixes Case 2 for you?
> https://github.com/bharathi-tenneti/operator-sdk/tree/issue%232622

@bharathi-tenneti  i'm getting this error, right after fully rendered chart
 ```
2020-04-09T00:28:34.440Z	ERROR	controller-runtime.controller	Reconciler error	{"controller": "k10-controller", "request": "kasten/k10", "error": "Operation cannot be fulfilled on k10s.apik10.kasten.io \"k10\": the object has been modified; please apply your changes to the latest version and try again"}
github.com/go-logr/zapr.(*zapLogger).Error
	/repo/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/repo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:258
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/repo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/repo/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/repo/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/repo/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/repo/go/pkg/mod/k8s.io/apimachinery@v0.17.4/pkg/util/wait/wait.go:88
```

**Comment by depohmel** [comment]
good news that I can see Deployments in logs now.

**Comment by bharathi-tenneti** [comment]
@depohmel Great, so did I understand correctly, that your issue is  getting fixed when you use SDK repo from thins branch?
https://github.com/bharathi-tenneti/operator-sdk/tree/issue%232622

**Comment by depohmel** [comment]
> @depohmel Great, so did I understand correctly, that your issue is getting fixed when you use SDK repo from this branch?
> https://github.com/bharathi-tenneti/operator-sdk/tree/issue%232622
> Once you give confirmation, ill submit PR to operator-sdk repo, and this fix will be available in next release.

Yes, one of them :) there is still another error :(

**Comment by joelanford** [comment]
@depohmel That seems like a completely separate issue. Can you file a separate GitHub issue for that one?

**Comment by depohmel** [comment]
@joelanford @bharathi-tenneti https://github.com/operator-framework/operator-sdk/issues/2815

**Comment by jsm84** [comment]
OK I've found the bug in the chart that @depohmel is using. It's the exact case #2. There was a template file, v0services.yaml inside the chart that had the following lines:

```
---
{{ end -}}
{{/*aggregatedapis-svc can be exposed via gateway later is required.*/}}
---
```

I've removed the comment and the latter `---` from the file, and the operator no longer throws the error, and it renders/prints the chart to stdout as normal. @joelanford As you mentioned, I think the fix for helm-operator is to ignore input files without a valid GVK. This is definitely an issue with the chart that the standalone helm binary seems to ignore.
